framework/web/Request.php

```diff
@@ -842,8 +950,8 @@ public function setUrl($value)
      */
     protected function resolveRequestUri()
     {
-        if (isset($_SERVER['HTTP_X_REWRITE_URL'])) { // IIS
-            $requestUri = $_SERVER['HTTP_X_REWRITE_URL'];
+        if ($this->headers->has('X-Rewrite-Url')) { // IIS
```

Comment:
Note that any functionality could simply be implemented without this header by simply using Yii routing rules.

All the webserver needs to do is get the request to `index.php`. So even removing the support altogether could be defended.